homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,486 @@
+
+            self._attributes["internalIP"] = self.port_mapping["NewInternalClient"]
+            self._attributes["internalPort"] = self.port_mapping["NewInternalPort"]
+            self._attributes["externalPort"] = self.port_mapping["NewExternalPort"]
+            self._attributes["protocol"] = self.port_mapping["NewProtocol"]
+            self._attributes["description"] = self.port_mapping[
+                "NewPortMappingDescription"
+            ]
+        except FritzConnectionException:
+            _LOGGER.error(
+                "Authorization Error: Please check the provided credentials and verify that you can log into the web interface"
+            )
+            self._is_available = False
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error("Could not get %s state", self.name, exc_info=True)
+            self._is_available = False
+
+    async def _async_handle_port_switch_on_off(self, turn_on: bool) -> bool:
+
+        self.port_mapping["NewEnabled"] = "1" if turn_on else "0"
+        try:
+            self.hass.async_add_executor_job(
+                lambda: self.fritzbox_tools.connection.call_action(
+                    self.connection_type, "AddPortMapping", **self.port_mapping
+                )
+            )
+        except FritzSecurityError:
+            _LOGGER.error(
+                "Authorization Error: Please check the provided credentials and verify that you can log into the web interface",
+                exc_info=True,
+            )
+        except FritzConnectionException:
+            _LOGGER.error(
+                "Home Assistant cannot call the wished service on the FRITZ!Box",
+                exc_info=True,
+            )
+            return False
+
+        return True
+
+
+class FritzBoxDeflectionSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools PortForward switch."""
+
+    def __init__(self, fritzbox_tools, device_friendly_name, dict_of_deflection):
+        """Init Fritxbox Deflection class."""
+        self.fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self.dict_of_deflection = dict_of_deflection
+        self._attributes = {}
+        self.id = int(self.dict_of_deflection["DeflectionId"])
+
+        switch_info = SwitchInfo(
+            description=f"Call deflection {self.id}",
+            friendly_name=device_friendly_name,
+            icon="mdi:phone-forward",
+            type=SWITCH_TYPE_DEFLECTION,
+            callback_update=self._async_fetch_update,
+            callback_switch=self._async_switch_on_off_executor,
+        )
+        super().__init__(self.fritzbox_tools, switch_info)
+
+    async def _async_fetch_update(self):
+        """Fetch updates."""
+
+        try:
+            resp = await self.hass.async_add_executor_job(
+                lambda: self.fritzbox_tools.connection.call_action(
+                    "X_AVM-DE_OnTel:1", "GetDeflections"
+                )
+            )
```

Comment:
```suggestion
            resp = await self.hass.async_add_executor_job(
                self.fritzbox_tools.connection.call_action, "X_AVM-DE_OnTel:1", "GetDeflections"
            )
```

You shouldn't need a `lambda` here if you pass the function, and then the args